save-backend/src/main/kotlin/com/saveourtool/save/backend/configs/WebSecurityConfig.kt

```diff
@@ -4,140 +4,34 @@
-            .permitAll()
-            // resources for frontend
-            .pathMatchers("/*.html", "/*.js*", "/*.css", "/img/**", "/*.ico", "/*.png", "/particles.json")
-            .permitAll()
-    }
-        .and()
-        .run {
-            authorizeExchange()
-                .pathMatchers("/**")
-                .authenticated()
-        }
-        .and()
-        .run {
-            // FixMe: Properly support CSRF protection https://github.com/saveourtool/save-cloud/issues/34
-            csrf().disable()
-        }
-        .addFilterBefore(
-            AuthenticationWebFilter(authenticationManager).apply {
-                setServerAuthenticationConverter(CustomAuthenticationBasicConverter())
-            },
-            SecurityWebFiltersOrder.HTTP_BASIC,
-        )
-        .exceptionHandling {
-            it.authenticationEntryPoint(
-                HttpStatusServerEntryPoint(HttpStatus.UNAUTHORIZED)
-            )
-        }
-        .logout()
-        .disable()
-        .formLogin()
-        .disable()
-        .build()
-
-    fun roleHierarchy(): RoleHierarchy = mapOf(
-        Role.SUPER_ADMIN to listOf(Role.ADMIN, Role.OWNER, Role.VIEWER),
-        Role.ADMIN to listOf(Role.OWNER, Role.VIEWER),
-        Role.OWNER to listOf(Role.VIEWER),
-    )
-        .mapKeys { it.key.asSpringSecurityRole() }
-        .mapValues { (_, roles) -> roles.map { it.asSpringSecurityRole() } }
-        .let(RoleHierarchyUtils::roleHierarchyFromMap)
-        .let {
-            RoleHierarchyImpl().apply { setHierarchy(it) }
-        }
-
-    @PostConstruct
-    fun postConstruct() {
-        defaultMethodSecurityExpressionHandler.setRoleHierarchy(roleHierarchy())
-    }
-
-    companion object {
-        /**
-         * These endpoints will have `permitAll` enabled on them. We can't selectively put `@PreAuthorize("permitAll")` in the code,
-         * because it won't allow us to configure authenticated access to all other endpoints by default.
-         * Or we can use custom AccessDecisionManager later.
-         */
-        internal val publicEndpoints = listOf(
-            "/error",
-            // `CollectionView` is a public page
-            "/api/$v1/projects/not-deleted",
-            "/api/$v1/awesome-benchmarks",
-            "/api/$v1/check-git-connectivity-adaptor",
-            // `OrganizationView` is a public page
-            // fixme: when we will want to make organizations accessible for everyone, wi will need to add more endpoints here
-            "/api/$v1/organizations/**",
-            "/api/$v1/projects/get/projects-by-organization",
-            // `ContestListView` and `ContestView` are public pages
-            "/api/$v1/contests/*",
-            "/api/$v1/contests/active",
-            "/api/$v1/contests/finished",
-            "/api/$v1/contests/*/public-test",
-            "/api/$v1/contests/*/scores",
-            "/api/$v1/contests/*/*/best",
-        )
-    }
-}
+class BackendWebSecurityConfig
```

Comment:
why do we need it? can we just add use `org.springframework.boot.autoconfigure.EnableAutoConfiguration=`?

Reply:
Seems, that in this case it does not see `WebSecurityConfig`